docs/research/461-model-providers-design.md

```diff
@@ -0,0 +1,729 @@
+        model: String,
+
+        /// Optional description
+        #[arg(short, long)]
+        description: Option<String>,
+
+        /// Path to JSON file with provider-specific settings
+        #[arg(long, value_name = "FILE")]
+        settings: Option<PathBuf>,
+
+        /// Rate limit: requests per second
+        #[arg(long)]
+        rate_limit: Option<u32>,
+    },
+
+    /// Update an existing model configuration
+    Update {
+        /// Configuration ID to update
+        config_id: String,
+
+        /// New name
+        #[arg(short, long)]
+        name: Option<String>,
+
+        /// New description
+        #[arg(short, long)]
+        description: Option<String>,
+
+        /// Path to JSON file with updated settings
+        #[arg(long, value_name = "FILE")]
+        settings: Option<PathBuf>,
+
+        /// New rate limit: requests per second
+        #[arg(long)]
+        rate_limit: Option<u32>,
+    },
+
+    /// Delete a model configuration
+    Delete {
+        /// Configuration ID to delete
+        config_id: String,
+
+        /// Skip confirmation prompt
+        #[arg(short = 'y', long)]
+        yes: bool,
+    },
+}
+```
+
+## 2. Flag Naming Conventions
+
+### 2.1 Standard Flags (Inherited from Existing Patterns)
+
+| Flag | Type | Default | Purpose | Precedent |
+|------|------|---------|---------|-----------|
+| `--limit` | u32 | 20 | Pagination limit | prompt.rs:20, graph.rs:18 |
+| `--offset` | u32 | 0 | Pagination offset | prompt.rs:24, graph.rs:22 |
+| `--organization-id` | String | None | Scope to organization | prompt.rs:28 |
+| `--workspace-id` | String | None | Scope to workspace | prompt.rs:32 |
+| `-f, --format` | String | table | Output format (global flag) | main.rs:25 |
+
+### 2.2 Model-Config-Specific Flags
+
+| Flag | Short | Type | Default | Purpose |
+|------|-------|------|---------|---------|
+| `--name` | `-n` | String | Required | Configuration name |
+| `--provider` | `-p` | String | Required (create) | Provider type filter/value |
+| `--model` | `-m` | String | Required (create) | Model identifier |
+| `--description` | `-d` | String | None | Human-readable description |
+| `--settings` | | PathBuf | None | Path to settings JSON file |
+| `--rate-limit` | | u32 | None | Requests per second |
+| `--name-contains` | | String | None | Name filter (list operation) |
+| `--yes` | `-y` | bool | false | Skip confirmation |
+
+**Design principles:**
+1. **Required vs optional**: Only essential fields are required; all configuration is optional
+2. **Short flags**: Only for frequently used flags (`-n`, `-p`, `-m`, `-d`)
+3. **Naming**: Use snake_case for multi-word flags (`--name-contains`, `--rate-limit`)
+4. **File inputs**: Use `--settings` (not `--settings-file`) following `--schema` precedent (prompt.rs:100)
+5. **Boolean flags**: No value required, presence = true (prompt.rs:36, graph.rs:89)
+
+### 2.3 Provider Values
+
+Supported provider values (based on scout research):
```

Comment:
If the SDK supports a value that is not in this list, we should not prevent the user from using that value, even if we do not know the full list of provider names. We should provide first class support for those that are listed but not prevent using the CLI from defining additional provider types if the information is available to the caller.

Reply:
Fixed in commit 189eafa. Clarified that:

- **CLI accepts ANY string value** for `--provider` (not restricted to the table)
- **"Known providers"** = first-class support (verified structures, used in docs/examples)
- **Extensibility**: Future providers or custom configurations work without CLI changes
- **Validation at API boundary**: CLI is thin passthrough layer

Added explicit design principle: "The CLI is a thin layer that passes user intent to the API. Provider-specific validation belongs at the API boundary, not in the CLI tool."

This ensures the CLI remains flexible and doesn't need updates when LangSmith adds new provider integrations.